### PR TITLE
Eliminate temporarily leaking goroutine in reconnectDownedHosts

### PR DESCRIPTION
**Pull request by malandrew**
closes: https://github.com/gocql/gocql/issues/774

### CODE REVIEW

File: session.go [hunk 1]
```diff
@@ -210,25 +213,31 @@ func NewSession(cfg ClusterConfig) (*Session, error) {
...
-
-		hosts := s.ring.allHosts()
-
-		// Print session.ring for debug.
-		if gocqlDebug {
-			buf := bytes.NewBufferString("Session.ring:")
-			for _, h := range hosts {
-				buf.WriteString("[" + h.Peer() + ":" + h.State().String() + "]")
+	s.quit = make(chan struct{})
+	s.reconnectTicker = time.NewTicker(intv)
```
Comment (Zariel) [inline session.go]: we dont need this on the session, also Stop it with defer

File: session.go [hunk 2]
```diff
@@ -210,25 +213,31 @@ func NewSession(cfg ClusterConfig) (*Session, error) {
...
-		time.Sleep(intv)
-
-		hosts := s.ring.allHosts()
-
-		// Print session.ring for debug.
-		if gocqlDebug {
-			buf := bytes.NewBufferString("Session.ring:")
-			for _, h := range hosts {
-				buf.WriteString("[" + h.Peer() + ":" + h.State().String() + "]")
+	s.quit = make(chan struct{})
```
Comment (Zariel) [inline session.go]: move this to be made when the session is created

**Comment by Zariel** [comment]
Also, if not already could you add yourself to the AUTHORS file

**Comment by malandrew** [comment]
@Zariel this is now ready to go.

**Comment by Zariel** [comment]
Great thanks for the contribution!

**Comment by malandrew** [comment]
@Zariel thanks for being a responsive maintainer.

**Pull request merged**